### PR TITLE
[BEAM-3118] Fix thread leaks in ElasticsearchIOTest.testWriteWith* when using DoFnTester

### PR DESCRIPTION
**Pull request by echauchot**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [X] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [X] Each commit in the pull request should have a meaningful subject line and body.
 - [X] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [X] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [X] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [X] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
DoFnTester is configured in CLONE_ONCE cloning behavior. So we need to explicitly call DoFnTester.close() so that doFn.tearDown() could be called and the ES Resclient properly closed causing the http leaked threads to be properly terminated.
R: @jkff

### CODE REVIEW

**Comment by echauchot** [comment]
@jkff try/catch in `TestWriteWithErrors` could be removed and `ExceptedException` put back with https://issues.apache.org/jira/browse/BEAM-3158

File: sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestCommon.java [hunk 1]
```diff
@@ -197,8 +191,18 @@ void testWriteWithErrors() throws Exception {
...
-    expectedException.expectMessage(
+    // need to avoid flow interruption to close the DoFnTester
+    Exception raisedException = null;
+    try {
+      // inserts into Elasticsearch
+      fnTester.processBundle(input);
+    } catch (IOException exception) {
+      raisedException = exception;
+    }
+    fnTester.finishBundle();
```
Comment (jkff) [inline sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestCommon.java]: * finishBundle shouldn't be necessary - processBundle should have called it. Does it not work without this?
* Can you use try-with-resources for closing (try (DoFnTester...))?

Comment (echauchot) [reply sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestCommon.java]: - indeed it works without the `finishBundle`, thanks for pointing out
- yes better indeed to use try with resources. And it is compatible with `ExpectedException` junit rule, thanks.

**Comment by echauchot** [comment]
leave the desactivation of the thread leaks checker because there is other unrelated thread leaks (https://issues.apache.org/jira/browse/BEAM-3119)

**Comment by echauchot** [comment]
@jkff I addressed all your comments PTAL

**Pull request closed without merging**